port/espressif/esp/src/hal/time.zig

```diff
@@ -0,0 +1,62 @@
+const std = @import("std");
+const microzig = @import("microzig");
+const time = microzig.drivers.time;
+
+// NOTE: esp32c3 specific
+const SYSTIMER = microzig.chip.peripherals.SYSTIMER;
+
+pub fn get_time_since_boot() time.Absolute {
+    return @enumFromInt(get_time());
+}
+
+pub fn sleep_ms(time_ms: u32) void {
+    sleep_us(time_ms * 1000);
+}
+
+pub fn sleep_us(time_us: u64) void {
+    const end_time = time.make_timeout_us(get_time_since_boot(), time_us);
+    while (!end_time.is_reached_by(get_time_since_boot())) {}
+}
+
+fn get_systimer_counter(counter_id: u32) u64 {
+    // TODO: We need to put it somewhere else
```

Comment:
where should this go?

Reply:
maybe `hal.init_sequence`? If so, every function that overrides it will have to be updated (e.g. the clock example)

Reply:
I did this